examples/flag_locations/BUILD.bazel

```diff
@@ -0,0 +1,23 @@
+load(
+    "@rules_rust//rust:rust.bzl",
+    "rust_test",
+)
+
+# generate a file containing cfg flags
+genrule(
```

Comment:
Let me start with saying I know it's a lot of boilerplate to write, but I personally would write a unit test under `//test/unit` (https://docs.bazel.build/versions/main/skylark/testing.html#for-testing-rules). This is an existing unit test that verifies args: https://github.com/bazelbuild/rules_rust/blob/main/test/unit/use_libtest_harness/use_libtest_harness_test.bzl.

Reply:
Extra points for adding tests for that logic adding `pwd` prefix, but I don't insist. I think you'll be suffering way too much already.

Reply:
This actually wouldn't pass without adding `${pwd}` to the expected argument string. Confusingly,  `assert_argv_contains` calls:

`"Expected {args} to contain {flag}".format(args = action.argv, flag = flag)`

on error. I think this expands `${pwd}` to the empty string under test, so it took me a while to figure out that I needed `@${pwd}/bazel-out/...` instead of `@/bazel-out/...`

Reply:
I'm going to leave the example as well. They both cover the same code, but I find the analysis test difficult to read, and I know more about Bazel than most people on the planet :)

Reply:
Sounds good!